Refactor Terms step of Wallet_Activate into `AcceptWalletTerms`

francoisl opened the pull request:
cc @MariaHCD 

Hold until https://github.com/Expensify/Web-Expensify/pull/34588 is at least on staging.

### Details
1. Use the new API command `AcceptWalletTerms` when the user accepts the terms of the Expensify Wallet, instead of using the deprecated `Wallet_Activate` command
2. Revamp the `ActivateStep` component to be prettier when the user finishes setting up a Gold wallet.

This PR does **not** implement the Red Brick Road for KYC errors. This will come in follow-up PRs.

### Fixed Issues
Part of https://github.com/Expensify/Expensify/issues/218502

### Tests

#### Pre-requisites:
1. Have an account with a pending wallet (all new accounts have a pending wallet created automatically)
3. Find the wallet's `bankAccountID` of that account in auth, and then run (replace the `bankAccountID` at the end):
```sql
UPDATE bankAccounts SET additionalData = JSON_SET(COALESCE(NULLIF(additionalData, ''), '{}'), '$.currentStep', 'TermsStep') WHERE bankAccountID = XXX;
```

----

#### Tests
1. Navigate to `/enable-payments`
2. You'll see the "Terms and fees" page.
<img width="1503" alt="image" src="https://user-images.githubusercontent.com/2229301/185290799-72141f24-da58-4562-99c7-78a3a542492a.png">
3. Scroll down to the bottom, and make sure you can't click the "Enable Payments" button without checking both buttons
<img width="381" alt="image" src="https://user-images.githubusercontent.com/2229301/185923247-23de889f-be11-4118-b6a0-8cee613b8817.png">

4. Check both checkboxes, click the Enable Payments button

5. Since the wallet technically doesn't pass all identity checks at that point, it will be a Silver wallet, and you should see a "We're still reviewing your information. Please check back later."  message 

<img height="150" alt="image" src="https://user-images.githubusercontent.com/2229301/185291226-9ae6afb4-1dd3-438c-a1f6-8e748866c051.png"><img height="150" alt="image" src="https://user-images.githubusercontent.com/2229301/185680731-f32d8411-4584-4986-a0fd-c824ca1cb785.png">

6. Run the flow again but for a Gold wallet. Start by resetting and configuring the wallet as follows:

    1. Run the `validator:wallet` CLI Tool on the account
    2. In auth, run: `UPDATE bankAccounts SET additionalData = JSON_SET(COALESCE(NULLIF(additionalData, ''), '{}'), '$.currentStep', 'TermsStep') WHERE bankAccountID = <walletID>;`
    3. In NewDot, run the following in the console: `Onyx.merge('userWallet', {shouldShowWalletActivationSuccess:null})`
    
7. Navigate to  `/enable-payments` again. Check both boxes at the bottom, and click the button
9. Make sure you see the following Tada page

<img width="1292" alt="Screen Shot 2022-08-22 at 1 00 54 PM" src="https://user-images.githubusercontent.com/12268372/185882899-2b595b2d-9d77-4a1b-896e-4e5899c89124.png">

- [ ] Verify that no errors appear in the JS console

### PR Review Checklist
<!--
This is a checklist for PR authors & reviewers. Please make sure to complete all tasks and check them off once you do, or else Expensify has the right not to merge your PR!
-->
#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I ran the tests on **all platforms** & verified they passed on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The Contributor+ will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
8. Upload an image via copy paste
10. Verify a modal appears displaying a preview of that image
--->

- [ ] Verify that no errors appear in the JS console

### Screenshots
<!-- Add screenshots for all platforms tested. Pull requests won't be merged unless the screenshots show the app was tested on all platforms.-->

#### Web
<!-- Insert screenshots of your changes on the web platform-->

#### Mobile Web
<!-- Insert screenshots of your changes on the web platform (from a mobile browser)-->

#### Desktop
<!-- Insert screenshots of your changes on the desktop platform-->

#### iOS
<!-- Insert screenshots of your changes on the iOS platform-->

#### Android
<!-- Insert screenshots of your changes on the Android platform-->

francoisl committed:
Call new `AcceptWalletTerms` API command

francoisl committed:
Set up `shouldShowWalletActivationSuccess` user wallet prop in ActivateStep view

francoisl committed:
Merge main, fix conflicts

francoisl committed:
Upgrade `ActivateStep` component to show a Tada 🎉 message for Gold wallets

MariaHCD commented on src/pages/EnablePayments/ActivateStep.js:
```diff
@@ -21,24 +27,61 @@ const defaultProps = {
+    renderGoldWalletActivationStep() {
+        return (
+            <>
+                <View style={[styles.pageWrapper, styles.flex1, styles.flexColumn, styles.alignItemsCenter, styles.justifyContentCenter]}>
+                    <Icon
+                        src={Illustrations.TadaBlue}
+                        height={100}
+                        width={100}
+                        fill={defaultTheme.iconSuccessFill}
+                    />
+                    <View style={[styles.ph5]}>
+                        <Text style={[styles.mt5, styles.h1, styles.textAlignCenter]}>
+                            {this.props.translate('activateStep.activated')}
+                        </Text>
+                    </View>
+                </View>
+                <FixedFooter>
+                    <Button
+                        text={this.props.translate('common.continue')}
+                        onPress={PaymentMethods.continueSetup}
+                        style={[styles.mt4]}
+                        iconStyles={[styles.mr5]}
+                        success
+                    />
+                </FixedFooter>
```
This doesn't quite match the mockup from the doc :)

https://docs.google.com/document/d/1PYqKcBpjPxewL5OneCmqbMJNH8G3CNkqbJWqzHSaXhU/edit#bookmark=id.ss0zd8oyorgm

MariaHCD commented on src/pages/EnablePayments/ActivateStep.js:
```diff
@@ -21,24 +27,61 @@ const defaultProps = {
+                />
+                <View style={[styles.mh5, styles.flex1]}>
+                    {this.props.userWallet.tierName === CONST.WALLET.TIER_NAME.GOLD && this.renderGoldWalletActivationStep()}
+                    {this.props.userWallet.tierName === CONST.WALLET.TIER_NAME.SILVER && (
+                        <Text>{this.props.translate('activateStep.checkBackLater')}</Text>
+                    )}
```
Created an issue for a mockup for the `Check back later` screen here: https://github.com/Expensify/Expensify/issues/223443

Doesn't have to be part of this PR, can be a follow up improvement.

francoisl committed:
Display errors returned by response

francoisl committed:
Add submessage for gold wallet activated step

francoisl commented on src/libs/actions/Wallet.js:
```diff
@@ -377,6 +377,47 @@ function activateWallet(currentStep, parameters) {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: ONYXKEYS.USER_WALLET,
+            value: {
+                shouldShowWalletActivationSuccess: null,
```
I changed this to `null` instead of `false` so it gets removed, I figured it's no longer needed at that point.

francoisl commented on src/languages/es.js:
```diff
@@ -694,7 +694,8 @@ export default {
     activateStep: {
         headerTitle: 'Habilitar pagos',
-        activated: 'Su billetera Expensify está lista para usar.',
+        activatedTitle: 'Billetera Expensify lista!',
+        activatedMessage: 'Enhorabuena, tu billetera Expensify está lista para usar.',
```
Getting confirmation for these [here](https://expensify.slack.com/archives/C21FRDWCV/p1660882913085719).

MariaHCD commented on src/libs/actions/Wallet.js:
```diff
@@ -377,6 +377,47 @@ function activateWallet(currentStep, parameters) {
+ * @param {Boolean} parameters.hasAcceptedTerms
+ */
+function acceptWalletTerms(parameters) {
+    const optimisticData = [
```
Looks like we use walletTerms.loading here: https://github.com/Expensify/App/blob/e9d29b12e27b57138fb225b67054e508d3274766/src/pages/EnablePayments/TermsStep.js#L119

I guess we don't really need it since the activation success view will be optimistically shown?

francoisl replied:
Ah yeah, good point. Looks like it's the only place `loading` was used, so I'll just remove the prop entirely.

MariaHCD reviewed (commented):
LGTM. I think we just need those translations and then we should be good to go!

francoisl committed:
Update spanish translation for wallet pending validation

francoisl committed:
Remove `walletTerms.loading` prop that's no longer necessary

francoisl commented:
Updated and removed the hold.

One thing I'm not sure about is that I get an error in the JS console that seems to point to something unrelated, but it's also happening on `main`.

```
Warning: Failed prop type: Invalid prop `userWallet.availableBalance` of type `number` supplied to `AvatarWithIndicator`, expected `object`.
    at AvatarWithIndicator (webpack-internal:///./src/components/AvatarWithIndicator.js:69:23)
```
it points to the `availableBalance` prop in `userWalletPropTypes`, which is already defined as a number and is not required..? We can probably ignore for this PR but just checking.

francoisl committed:
Merge main, fix conflicts

francoisl commented:
Fixed the conflicts with your other PR we just merged, and also I'm putting this back on hold until https://github.com/Expensify/Web-Expensify/pull/34588 is on staging.

francoisl committed:
Merge branch 'main' of github.com:Expensify/App into francois-refactorAcceptWalletTerms

MariaHCD commented on src/pages/EnablePayments/userWalletPropTypes.js:
```diff
@@ -19,4 +19,7 @@ export default PropTypes.shape({
     /** Status of wallet - e.g. SILVER or GOLD */
     tierName: PropTypes.string,
+
+    /** Whether we should show the ActivateStep view success view after the user finished the KYC flow */
```
```suggestion
    /** Whether we should show the ActivateStep success view after the user finished the KYC flow */
```

MariaHCD commented on src/pages/EnablePayments/TermsStep.js:
```diff
@@ -112,11 +114,15 @@ class TermsStep extends React.Component {
+                            {_.last(_.values(errors))}
+                        </Text>
+                    )}
                     <Button
```
Let's replace the Button component with the FormAlertWithSubmitButton. It's more consistent with all other forms and we can get rid of [these lines
](https://github.com/Expensify/App/blob/86bf03a80b5709b702c4e334fc05679c6e05876e/src/pages/EnablePayments/TermsStep.js#L112-L120)

```js
const bool isButtonDisabled = !this.state.hasAcceptedDisclosure || !this.state.hasAcceptedPrivacyPolicyAndWalletAgreement;
...
<FormAlertWithSubmitButton
    buttonText={this.props.translate('termsStep.enablePayments')}
    onSubmit={() => {
        if (!this.state.hasAcceptedDisclosure
            || !this.state.hasAcceptedPrivacyPolicyAndWalletAgreement) {
            this.setState({error: true});
            return;
        }

        this.setState({error: false});
        BankAccounts.acceptWalletTerms({
            hasAcceptedTerms: this.state.hasAcceptedDisclosure
                && this.state.hasAcceptedPrivacyPolicyAndWalletAgreement,
        });
    }}
    isDisabled={isButtonDisabled || this.props.network.isOffline}
     message={this.state.error ? this.props.translate('common.error.acceptedTerms') : !_.isEmpty(errors) ? _.last(_.values(errors)) : ''}
    isAlertVisible={this.state.error || !_.isEmpty(errors)}
/>
```

<img width="945" alt="Screen Shot 2022-08-22 at 1 46 47 PM" src="https://user-images.githubusercontent.com/12268372/185892563-d282caff-20cf-4508-a74f-f18be95f8236.png">

MariaHCD commented on src/pages/EnablePayments/ActivateStep.js:
```diff
@@ -21,24 +27,64 @@ const defaultProps = {
+                <FixedFooter>
+                    <Button
+                        text={this.props.translate('common.continue')}
+                        onPress={PaymentMethods.continueSetup}
```
Haven't tested but just wanted to confirm if clicking this will lead the user back to the action they were previously taking i.e. paying via the wallet or transferring their wallet balance?

francoisl replied:
This is a huge pain to test on dev for some reason, but yes. It calls [this function](https://github.com/Expensify/App/blob/57d2d97809913cd029baea399e09155ef24dd916/src/components/KYCWall/BaseKYCWall.js#L86-L93), which at the bottom ends up calling the `onSuccessfulKYC()` prop method.

https://user-images.githubusercontent.com/2229301/185936994-0a904103-7edd-499b-b0ec-b102e1c72a60.mov

MariaHCD commented on src/pages/EnablePayments/ActivateStep.js:
```diff
@@ -21,24 +27,64 @@ const defaultProps = {
+                </View>
+                <FixedFooter>
+                    <Button
+                        text={this.props.translate('common.continue')}
```
Not a big deal but we can probably adjust this now or in a follow up PR.

```
The green success button will either state “Continue to payment” or “Continue to transfer” based on the action that we will be returning the user to.
```

https://docs.google.com/document/d/1PYqKcBpjPxewL5OneCmqbMJNH8G3CNkqbJWqzHSaXhU/edit#bookmark=id.8y4wjrbvjh9z

francoisl replied:
Yeah at the moment I don't see how we would do that so I don't mind if we do this in a follow-up PR.

francoisl committed:
PR comments

MariaHCD commented:
> One thing I'm not sure about is that I get an error in the JS console that seems to point to something unrelated, but it's also happening on `main`.

Yeah, I see this is happening for the `EnablePaymentsPage` and `AvatarWithIndicator` but I have no idea why 🤔 I think it was probably introduced [here](https://github.com/Expensify/App/pull/10322/files#diff-8fa4a1615ab5b94af2a237f796a30eab18b92254de85c86817a3b68d39c2f0ddR1) when I updated `userWalletPropTypes` and is unrelated to this PR so I think its okay if we ignore it for now. I will try to investigate a bit more. 

```
react.development.js?72d0:209 Warning: Failed prop type: Invalid prop `userWallet.availableBalance` of type `number` supplied to `EnablePaymentsPage`, expected `object`.
```

```
Warning: Failed prop type: Invalid prop `userWallet.availableBalance` of type `number` supplied to `AvatarWithIndicator`, expected `object`.
```

MariaHCD commented on src/pages/EnablePayments/ActivateStep.js:
```diff
@@ -21,24 +27,64 @@ const defaultProps = {
+                    shouldShowBackButton
+                    onBackButtonPress={() => Navigation.goBack()}
+                />
+                <View style={[styles.mh5, styles.flex1]}>
```
```suggestion
                <View style={styles.flex1}>
```

Let's remove the horizontal margin, it seems a little excessive in comparison to the mockup:


| Current      | Mockup | After removing mh5 |
| ----------- | ----------- |----------- |
|   ![Screen Shot 2022-08-22 at 6 59 50 PM](https://user-images.githubusercontent.com/12268372/185953440-0887d197-a3d0-44f6-9159-e015bcf444b9.png)   |   <img src="https://user-images.githubusercontent.com/12268372/178017051-572df5e7-eb0d-4d91-9f85-8fdbf81d01f4.png">    |  ![Screen Shot 2022-08-22 at 6 55 20 PM](https://user-images.githubusercontent.com/12268372/185953247-ad2ccf40-d995-492c-933e-b77e6e1d7a24.png) |

MariaHCD reviewed (commented):
Minor change but otherwise looks good to me!

francoisl committed:
Remove horizontal margin on button

Co-authored-by: Maria D'Costa <maria@expensify.com>

francoisl commented:
Updated, thanks for the reviews!

MariaHCD reviewed (approved):
LGTM!

francoisl commented:
Removed the Hold on this. I don't know if @Justicea83 you want to review? Otherwise Maria feel free to merge!

Justicea83 reviewed (approved):
Changes LGTM!

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/2913733257) to staging by @Justicea83 in version: 1.1.89-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅
